### PR TITLE
fix(pr-comments): remove PR id key constraint

### PR DESCRIPTION
**Pull request by cathteng**
[This issue](https://sentry.sentry.io/issues/4897265143/?) is occurring for the merged PR comment bot in which we attempt to comment on a PR that already has an open PR comment, and it errors saying that a comment already exists for that particular PR.

We have [this uniqueness constraint](https://github.com/getsentry/sentry/blob/f91048925f8ba28c35d91e19ebb078d35e44220a/src/sentry/models/pullrequest.py#L121) aka `("pull_request_id", "comment_type")` on the table to allow for up to one comment per comment type, but there is an additional existing constraint on the `pull_request` field (`sentry_pullrequest_comment_pull_request_id_key`) that has precedence because it limits comments to 1 per PR. ([source](https://redash.getsentry.net/queries/5891/source)).

We simply need to remove this constraint to allow multiple comments per PR.

### CODE REVIEW

File: src/sentry/models/pullrequest.py [hunk 1]
```diff
@@ -118,4 +118,4 @@ class PullRequestComment(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_pullrequest_comment"
-        unique_together = ("pull_request", "comment_type")
+        unique_together = (("pull_request", "comment_type"),)
```
Comment (cathteng) [inline src/sentry/models/pullrequest.py:121]: will this constraint become active once we remove the pull_request constraint? i'm not sure if i need to write some custom SQL operations to make this one the active one

File: src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        ("sentry", "0636_monitor_incident_env_resolving_index"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="pullrequestcomment",
+            name="pull_request",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                db_constraint=False,
+                on_delete=django.db.models.deletion.CASCADE,
```
Comment (markstory) [inline src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py:33]: The `on_delete` clause won't do anything without the constraint.

File: src/sentry/models/pullrequest.py (hunk 1 above)
Comment (markstory) [reply src/sentry/models/pullrequest.py:121]: No, there are no DDL operations in your migration to add the missing constraint. You could use a SQL operation to define the constraint if it doesn't exist. Using a simple DDL operation could fail if any of the environments/tenants have this constraint already.

File: src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py (hunk 2 above)
Comment (cathteng) [reply src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py:33]: so it won't cascade if the PR itself is deleted?

Comment (cathteng) [reply src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py:33]: does this mean i need to add some sort of custom deletion logic to delete `PullRequestComment` objects appropriately?

File: src/sentry/models/pullrequest.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ class PullRequestComment(Model):
     __relocation_scope__ = RelocationScope.Excluded
 
     external_id = BoundedBigIntegerField()
-    pull_request = FlexibleForeignKey("sentry.PullRequest")
+    pull_request = FlexibleForeignKey("sentry.PullRequest", db_constraint=False)
```
Comment (wedamija) [inline src/sentry/models/pullrequest.py:109]: This isn't a unique constraint, I don't understand why we need to remove it here?

Comment (cathteng) [reply src/sentry/models/pullrequest.py:109]: or i might be wrong and the unique_together constraint wasn't added in the first place?

Comment (cathteng) [reply src/sentry/models/pullrequest.py:109]: okay upon second glance i actually don't need to do this. we are missing the uniqueness constraint with pullrequest + comment_type so I can just add that

Comment (cathteng) [reply src/sentry/models/pullrequest.py:109]: i updated the PR

File: src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py (hunk 2 above)
Comment (cathteng) [reply src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py]: on second glance, i was querying for the uniqueness constraints incorrectly. i fixed the query and found the offending constraint

**Review by wedamija — APPROVED** [review]

**Review by markstory — APPROVED** [review]
Looks good, this will remove the constraint in any environment that it does exist so that a new constraint can be added cleanly.

**Pull request merged**